src/fides/api/tasks/__init__.py

```diff
@@ -20,10 +29,27 @@
 ]
 
 
+def log_retry_attempt(retry_state: RetryCallState) -> None:
+    """Log database connection retry attempts."""
+
+    logger.warning(
+        "Database connection attempt {} failed. Retrying in {} seconds...",
+        retry_state.attempt_number,
+        retry_state.next_action.sleep,  # type: ignore[union-attr]
+    )
+
+
 class DatabaseTask(Task):  # pylint: disable=W0223
     _task_engine = None
     _sessionmaker = None
 
+    @retry(
+        stop=stop_after_attempt(NEW_SESSION_RETRIES),
+        wait=wait_exponential(multiplier=1, min=1),
+        retry=retry_if_exception_type(OperationalError),
+        reraise=True,
+        before_sleep=log_retry_attempt,
+    )
```

Comment:
This will attempt to connect 5 times with an exponential backoff (2, 4, 8, 16 seconds between each attempt). The original error will be re-raised if the retries are successful. All attempts are shown in the logs.
```
2024-11-01 21:57:43.511 | WARNING  | fides.api.tasks:log_retry_attempt:34 - Database connection attempt 1 failed. Retrying in 1.0 seconds...
2024-11-01 21:57:44.516 | WARNING  | fides.api.tasks:log_retry_attempt:34 - Database connection attempt 2 failed. Retrying in 2.0 seconds...
2024-11-01 21:57:46.521 | WARNING  | fides.api.tasks:log_retry_attempt:34 - Database connection attempt 3 failed. Retrying in 4.0 seconds...
2024-11-01 21:57:50.531 | WARNING  | fides.api.tasks:log_retry_attempt:34 - Database connection attempt 4 failed. Retrying in 8.0 seconds...
2024-11-01 21:57:58.531 | WARNING  | fides.api.tasks:log_retry_attempt:34 - Database connection attempt 4 failed. Retrying in 16.0 seconds...
```
```

Reply:
Mind rewording your description as a code-comment?